service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/ControlledAzureKubernetesNamespaceResource.java

```diff
@@ -213,14 +215,17 @@ private List<Step> getCreateFederatedCredentialsSteps(FlightBeanBag flightBeanBa
   }
 
   @NotNull
-  private Step getGetManagedIdentityStep(FlightBeanBag flightBeanBag) {
+  private Step getGetManagedIdentityStep(
+      FlightBeanBag flightBeanBag, MissingIdentityBehavior missingIdentityBehavior) {
     return switch (getAccessScope()) {
       case ACCESS_SCOPE_SHARED -> new GetWorkspaceManagedIdentityStep(
-          flightBeanBag.getAzureConfig(),
-          flightBeanBag.getCrlService(),
           getWorkspaceId(),
-          flightBeanBag.getResourceDao(),
-          getManagedIdentity());
+          getManagedIdentity(),
+          missingIdentityBehavior,
+          new ManagedIdentityHelper(
```

Comment:
should this be a spring managed bean?

Reply:
I'm going to leave this as-is for simplicity; if this gets to more than a single callsite, it'd be a good idea.